### PR TITLE
Merge from OddBird fork.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
- Allow creation of Project-level scratch orgs (https://github.com/oddbird/Metecho/pull/940/)
- Allow creation of Epic-level scratch orgs (https://github.com/oddbird/Metecho/pull/685/)
- Allow creation of Task-level scratch orgs (https://github.com/oddbird/Metecho/pull/1018)
- Add Project intro modal and "Plan" guided tour (https://github.com/oddbird/Metecho/pull/772/)
  - This can be disabled globally by setting env var `ENABLE_WALKTHROUGHS=False`
- Allow assigning GitHub users directly to Task roles (https://github.com/oddbird/Metecho/pull/884/)
- Confirm before deleting scratch org with unretrieved changes (https://github.com/oddbird/Metecho/pull/1015)
- Add "Test" task status in tasks table (https://github.com/oddbird/Metecho/pull/1017)
- Style Django-served 404 page (https://github.com/oddbird/Metecho/pull/1016)
- Upgrade dependencies
- Continue adding Storybook component library (not yet deployed publicly)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,12 +73,6 @@ jscache/
 # Sphinx documentation
 docs/_build/
 
-# pyenv
-.python-version
```
Comment (davisagli) [inline .gitignore:79]: Why did you remove this? I use pyenv.

File: .python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.9.0
```
Comment (davisagli) [inline .python-version:1]: This file should not be in source control. I have mine configured to use a virtualenv named `metecho`

File: .yarnrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+yarn-path ".yarn/releases/yarn-1.22.5.cjs"
```
Comment (davisagli) [inline .yarnrc:5]: What is this?

File: config/settings/base.py [hunk 4]
```diff
@@ -509,6 +511,8 @@ def env(name, default=NoDefaultValue, type_=str):
 # New feature branch prefix:
 BRANCH_PREFIX = env("BRANCH_PREFIX", default=None)
 
+ENABLE_WALKTHROUGHS = env("ENABLE_WALKTHROUGHS", default=True, type_=boolish)
```
Comment (davisagli) [inline config/settings/base.py:514]: Don't bother changing it, but when we're putting a new feature behind a feature flag, it should probably be off by default.

File: .python-version (hunk 2 above)
Comment (jgerigmeyer) [reply .python-version:1]: This should only impact devs using pyenv, but I'll remove it.

File: .yarnrc (hunk 3 above)
Comment (jgerigmeyer) [reply .yarnrc:5]: The correct version of yarn is bundled with the repo (in `.yarn/releases`), as per yarn recommendations. Again, I can remove if you feel like that's an issue.

File: config/settings/base.py (hunk 4 above)
Comment (jgerigmeyer) [reply config/settings/base.py:514]: Makes sense. I went this way because I'm not used to using feature flags in this project and I was thinking of it as a temp workaround while we flesh out the other walkthroughs -- but I think you're right. 👍

File: .yarnrc (hunk 3 above)
Comment (davisagli) [reply .yarnrc:5]: Where can I read more about that recommendation? My initial reaction is that I don't love adding a large file to the repository if we don't need to, but I'd like to understand the argument in favor before deciding.

Comment (jgerigmeyer) [reply .yarnrc:5]: Yeah, the argument is basically "so users don't need to worry about local yarn versions". https://classic.yarnpkg.com/en/docs/cli/policies

Comment (davisagli) [reply .yarnrc:5]: It makes some sense; let's give it a try.

**Review by davisagli — APPROVED** [review]
